pkg/controller/core/workload_controller.go

```diff
@@ -127,6 +127,21 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	ctx = ctrl.LoggerInto(ctx, log)
 	log.V(2).Info("Reconciling Workload")
 
+	if eviction := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadEvicted); eviction != nil && eviction.Status == metav1.ConditionTrue &&
+		eviction.Reason == kueue.WorkloadEvictedByPodsReadyTimeout {
```

Comment:
I don't think we need to differentiate the reason here.
Can we just copy the eviction reason and message into the admission condition?

Reply:
I don't think we should make a difference on how Preemption and Eviction is treated here. Both modes are not immediate and potentially have to the job to be inactive.